android/app/src/main/java/org/openbot/env/UsbConnection.java

```diff
@@ -53,29 +52,22 @@ public UsbConnection(Context context, int baudRate) {
         PendingIntent.getBroadcast(this.context, 0, new Intent(ACTION_USB_PERMISSION), 0);
   }
 
-  private UsbSerialInterface.UsbReadCallback callback =
-      new UsbSerialInterface.UsbReadCallback() {
-        @Override
-        public void onReceivedData(byte[] data) {
-          try {
-            String dataUtf8 = new String(data, "UTF-8");
-            buffer += dataUtf8;
-            int index;
-            while ((index = buffer.indexOf('\n')) != -1) {
-              final String dataStr = buffer.substring(0, index).trim();
-              buffer = buffer.length() == index ? "" : buffer.substring(index + 1);
-
-              AsyncTask.execute(
-                  new Runnable() {
-                    @Override
-                    public void run() {
-                      onSerialDataReceived(dataStr);
-                    }
-                  });
-            }
-          } catch (UnsupportedEncodingException e) {
-            LOGGER.e("Error receiving USB data");
-          }
+  private final UsbSerialInterface.UsbReadCallback callback =
+      data -> {
```

Comment:
What happens on error, since try/catch is removed?

Reply:
It was an android studio recommendation that this catch block would never be called. I'll add it back. Better safer.